src/dotnet/Chat.UI.Blazor/Services/ChatUIStateSync.cs

```diff
@@ -113,18 +120,62 @@ async ValueTask<bool> IsLanguageChanged()
             return isLanguageChanged;
         }
 
-        Task SyncRecorderState()
-            => BackgroundTask.Run(async () => {
-                if (recorderState != null) {
+        Task SyncRecorderState() => RestartRecording(recorderState != null, recordingChatId, cancellationToken);
+    }
+
+    [ComputeMethod]
+    protected virtual async Task<(Symbol ChatId, long LastEntryId)> GetLastChatEntry(CancellationToken cancellationToken)
+    {
+        var recordingChatId = await ChatUI.RecordingChatId.Use(cancellationToken).ConfigureAwait(false);
+        if (recordingChatId.IsEmpty)
+            return (recordingChatId, 0);
+
+        var (_, end) = await Chats.GetIdRange(Session, recordingChatId, ChatEntryType.Text, cancellationToken).ConfigureAwait(false);
```

Comment:
1) `ChatEntryType.Audio`?
2) If we really want to track new messages from a given user, maybe it makes sense to add `GetIdRange(session, chatId, entryType, authorId, ct)`?

And 3)... Honestly, I'd rather do this in the audio processing pipeline, not here. Here it looks more like a misuse of a feature - i.e. using what we write for the feedback loop instead of just stopping the recording in the right place.

Reply:
There are a few reasons to do this in our audio pipeline:
- All such code should be in one place. Spreading across the codebase & basing it on completely different (and even unnecessary) things is deeply wrong.
- We anyway need some ticks in our audio pipeline - at least to detect audio issues, but also to run such actions as detecting idle recordings.

Reply:
Please disregard last sentences about audio pipeline - let's merge the PR